### PR TITLE
[fix](schema scanner)change schema_scanner::get_next_row to get_next_block

### PR DESCRIPTION
**Pull request by Cai-Yao**
# Proposed changes

Issue Number: close #xxx

## Problem summary

delete schema scanner::get_next_row，change to get_to_block

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/vschema_scan_node.cpp [hunk 1]
```diff
@@ -258,73 +262,112 @@ Status VSchemaScanNode::get_next(RuntimeState* state, vectorized::Block* block,
     std::vector<vectorized::MutableColumnPtr> columns(_slot_num);
     bool schema_eos = false;
 
-    do {
-        bool mem_reuse = block->mem_reuse();
-        DCHECK(block->rows() == 0);
+    if (_schema_scanner->type() == TSchemaTableType::SCH_TABLES) {
+        do {
+            bool mem_reuse = block->mem_reuse();
```
Comment (yiguolei) [inline be/src/vec/exec/vschema_scan_node.cpp:267]: do not use mem reuse here, just call block->clear() to clear all columns in block and then reinsert columns.

File: be/src/exec/schema_scanner/schema_tables_scanner.cpp [hunk 2]
```diff
@@ -52,7 +55,9 @@ SchemaTablesScanner::SchemaTablesScanner()
         : SchemaScanner(_s_tbls_columns,
                         sizeof(_s_tbls_columns) / sizeof(SchemaScanner::ColumnDesc)),
           _db_index(0),
-          _table_index(0) {}
+          _table_index(0) {
```
Comment (yiguolei) [inline be/src/exec/schema_scanner/schema_tables_scanner.cpp:58]: _table_index is useless

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.48 seconds
 load time: 633 seconds
 storage size: 17123553794 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230109051053_clickbench_pr_75856.html

File: be/src/exec/schema_scan_node.cpp [hunk 3]
```diff
@@ -279,7 +279,7 @@ Status SchemaScanNode::get_next(RuntimeState* state, RowBatch* row_batch, bool*
             return Status::OK();
         }
 
-        RETURN_IF_ERROR(_schema_scanner->get_next_row(_src_tuple, _tuple_pool.get(), &scanner_eos));
+        // RETURN_IF_ERROR(_schema_scanner->get_next_row(_src_tuple, _tuple_pool.get(), &scanner_eos));
```
Comment (yiguolei) [inline be/src/exec/schema_scan_node.cpp:282]: just remove the code

File: be/src/exec/schema_scanner/schema_rowsets_scanner.cpp [hunk 4]
```diff
@@ -97,80 +82,127 @@ Status SchemaRowsetsScanner::get_all_rowsets() {
...
-    RowsetSharedPtr rowset = rowsets_[rowsets_idx_];
+Status SchemaRowsetsScanner::get_next_block(vectorized::Block* block, bool* eos) {
+    if (!_is_init) {
+        return Status::InternalError("Used before initialized.");
+    }
+    if (nullptr == block || nullptr == eos) {
+        return Status::InternalError("input pointer is nullptr.");
+    }
+
+    *eos = true;
```
Comment (yiguolei) [inline be/src/exec/schema_scanner/schema_rowsets_scanner.cpp:93]: rowsets 这个表不一定可以一个block 一下子装进去的，可能有10000000个rowset

File: be/src/exec/schema_scanner.cpp [hunk 5]
```diff
@@ -231,4 +232,161 @@ Status SchemaScanner::create_tuple_desc(ObjectPool* pool) {
     return Status::OK();
 }
 
+Status SchemaScanner::fill_dest_column(vectorized::Block* block, void* data,
+                                       const SlotDescriptor* slot_desc) {
+    if (!block->has(slot_desc->col_name())) {
+        return Status::OK();
+    }
+    vectorized::IColumn* col_ptr = const_cast<vectorized::IColumn*>(
```
Comment (yiguolei) [inline be/src/exec/schema_scanner.cpp:240]: 这里你不要直接使用裸指针， 尽量使用 ColumnPtr 这种类似智能指针的结构

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**